slack/src/main/java/com/hazelcast/jet/contrib/slack/SlackSinks.java

```diff
@@ -0,0 +1,170 @@
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.apache.http.conn.HttpHostConnectException;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Contains the methods to create Slack sinks.
+ */
+public final class SlackSinks {
+
+    private static final String URL = "https://slack.com/api/chat.postMessage";
+
+    private static final int RETRY_COUNT = 5;
+
+    private SlackSinks() {
+    }
+
+    /**
+     * Creates a sink to send messages to requested channel id.
+     * Here the sink will be created with the default name as slack-sink.
+     *
+     * @param accessToken String Bearer token to authenticate the slack web api requests
+     * @param slackChannelId String Unique channel id to send messages to the slack channel.
+     * @return
+     */
+    public static Sink sink(String accessToken, String slackChannelId) {
+        return sink("slack-sink", accessToken, slackChannelId);
+    }
+
+    /**
+     * Creates a slack sink to send messages to the slack channel with the given slack channel name.
+     * Internally apache {@link org.apache.http.client.HttpClient} is used with
+     * custom {@link HttpRequestRetryHandler} to post the messages to slack channels.
+     * Following error causes {@link ConnectTimeoutException}, {@link HttpHostConnectException}, {@link IOException}
+     * are handled through custom retry handler.The max retry count has been configured to 5.
+     * Other error causes will lead to the breakage of the jet pipeline.
+     *
+     * @param accessToken String Bearer token to authenticate the slack web api requests
+     * @param slackChannelId String Unique channel id to send messages to the slack channel.
+     * @param slackChannelName String channel name to be used as sink name
+     * @return
+     */
+    public static Sink sink(String accessToken, String slackChannelId, String slackChannelName) {
+        return SinkBuilder.sinkBuilder(slackChannelName,
+                ctx -> new SlackContext(() -> HttpClients.custom().setRetryHandler(httpRetryHandler())
+                        .build(), accessToken, slackChannelId))
+                .<String>receiveFn((ctx, item) -> ctx.receiveFn(item)).destroyFn(ctx -> ctx.destroy()).build();
+    }
+
+    private static HttpRequestRetryHandler httpRetryHandler() {
+        return (exception, executionCount, context) -> {
+            if (executionCount >= RETRY_COUNT) {
+                // Do not retry if over max retry count
+                return false;
```

Comment:
I think we can retry forever. If you have the slack app, it also retries forever.

Reply:
I thought retrying forever might cause issues related to stability. Now I updated the retry count to Integer.max value instead of 3 times.